clients/cobol-lsp-vscode-extension/src/services/snippetcompletion/cobolSnippets.json

```diff
@@ -67,15 +109,80 @@
     "description": "ALTER TO"
   },
+  "AREA": {
+    "prefix": "AREA",
+    "body": ["${1|SAME,SAME SORT,SAME RECORD|} ${2|AREA,AREAS|}"],
+    "description": "The AREA/AREAS keywords pertain to the reuse of main storage. (ie. They force z/OS to use the same main storage area for performing internal file processing.) The three instances are syntax checked, but have no effect on the program at run-time."
+  },
   "ASCENDING": {
     "prefix": "ASCENDING",
-    "body": ["ASCENDING ${1:item}"],
+    "body": [
+      "01 ${3:coboltbl}",
+      "\t05 ${4:TBL-ROW} OCCURS ${5:numLiteral} TIMES",
+      "\t\tASCENDING ${1:field-1} ${2:field-2}.",
+      "$0"
+    ],
     "description": "ASCENDING item"
   },
-  "BEFORE": {
-    "prefix": "BEFORE",
-    "body": ["BEFORE ${1:id}"],
-    "description": "BEFORE literal"
+  "ASSIGN": {
+    "prefix": "ASSIGN",
+    "body": [
+      "FILE-CONTROL",
+      "\tSELECT ${2:rptfile}",
+      "\t\tASSIGN TO ${1:rptfile-DD}",
+      "\t\tORGANIZATION IS ${3|INDEXED,RELATIVE,SEQUENTIAL,LINE-SEQUENTIAL|}",
+      "\t\tACCESS MODE IS ${4|SEQUENTIAL,RANDOM,DYNAMIC|}",
+      "\t\tFILE STATUS IS ${5:statusVariable}.",
+      "$0"
+    ],
+    "description": "ASSIGN item"
+  },
+  "BEFORE ADVANCING": {
+    "prefix": "BEFORE ADVANCING",
+    "body": [
+      "WRITE ${1:01-OUTPUT-RECORD} BEFORE ADVANCING ${2:numLiteral} LINES",
+      "END-WRITE.",
+      "IF ${3:OUTPUT-RECORD-FILE-STATUS-CODE} NOT EQUAL 0 ${4:I/O-Error-Logic}",
+      "END-IF."
+    ],
+    "description": "The WRITE BEFORE phrase in the PROCEDURE DIVISION gives the COBOL programmer control over file (most often report file) processing pagination. Note that OUTPUT-RECORD-FILE-STATUS-CODE must be defined in the file FD and declared as a two-byte (PIC XX) field in the WORKING-STORAGE. https://publibfp.dhe.ibm.com/epubs/pdf/igy6lr40.pdf "
+  },
+  "BINARY": {
+    "prefix": "BINARY",
+    "body": ["BINARY ${1:id}"],
+    "description": "BINARY literal"
+  },
+  "BIT TO CHAR": {
+    "prefix": "BIT-TO-CHAR",
+    "body": [
+      "77 ${1:BIT-FLD} PIC X(16).",
+      "\tMOVE '${2:0000000000000000}' TO ${1:BIT-FLD}.",
+      "\tFUNCTION BIT-TO-CHAR(${1:BIT-FLD})."
+    ],
+    "description": "Calls BIT-TO-CHAR to obtain a string that corresponds to the bit pattern stored in the PIC X field."
+  },
+  "BLANK WHEN ZERO": {
+    "prefix": "BLANK",
+    "body": ["77 ${1:variableName} PIC S9(5)V99 BLANK WHEN ZERO."],
+    "description": "BLANK WHEN ZERO specifies that an elementary field will contains spaces when its value is zero."
+  },
+  "BLOCK": {
+    "prefix": "BLOCK",
+    "body": ["BLOCK CONTAINS 0 RECORDS"],
+    "description": "BLOCK CONTAINS clause (in the DATA DIVISION, file FD) dictates the size of the physical (I/O) records. By coding BLOCK CONTAINS 0 z/os will choose the optimal \"blocking factor\" for most efficient file I/O - taking into account channels, and DASD along with the file specifics."
```

Comment:
The BLOCK CONTAINS clause in the FD entry in the DATA DIVISION dictates the size of the physical (I/O) records. If you specify BLOCK CONTAINS 0, z/OS chooses the optimal \"blocking factor\" for most efficient file I/O - taking into account channels and DASD along with the file specifics.